native/packages/react-native-bpk-component-phone-input/src/BpkPhoneNumberInput.js

```diff
@@ -0,0 +1,66 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/* @flow */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import BpkTextInput, {
+  type BpkTextInputProps,
+} from 'react-native-bpk-component-text-input';
+import { CODE_PROP_TYPES, type RenderFlag } from './common-types';
+import BpkDialingCodeAccessoryView, {
+  type BpkDialingCodeAccessoryViewProps,
+} from './BpkDialingCodeAccessoryView';
+
+export type Props = {
+  ...$Exact<BpkTextInputProps>,
+  dialingCodeData: BpkDialingCodeAccessoryViewProps,
+  renderFlag: RenderFlag,
+};
+
+const BpkPhoneNumberInput = (props: Props) => {
+  const { renderFlag, dialingCodeData, editable, ...rest } = props;
+  const flag = renderFlag(dialingCodeData);
+
+  return (
+    <BpkTextInput
+      editable={editable}
+      {...rest}
+      accessoryView={
+        <BpkDialingCodeAccessoryView
+          editable={editable}
+          flag={flag}
+          {...props.dialingCodeData}
+        />
+      }
+    />
+  );
+};
+
+const { accessoryView, ...textInputPropTypes } = BpkTextInput.propTypes;
+const { accessoryView: a, ...textInputDefaultProp } = BpkTextInput.defaultProps;
```

Comment:
I can't see where `a` is being used ...

Reply:
Oh wait ... this is you trying to omit accessoryView from the prop types you wanna apply.

Reply:
Yeah I've updated this to give them clearer names i.e

```javascript
const {
  accessoryView: _thisIsIgnored,
  ...textInputPropTypes
} = BpkTextInput.propTypes;
const {
  accessoryView: _thisIsAlsoIgnored,
  ...textInputDefaultProp
} = BpkTextInput.defaultProps;
```

Open to other suggestion on how to do this

Reply:
I agree, that is better...